aea/skills/base.py

```diff
@@ -585,6 +585,9 @@ def __init__(
 
         self._contracts = {}  # type: Dict[str, Contract]
 
+        if self._skill_context:
```

Comment:
I prefer we keep the convention for explicit comparisons: `if self._skill_context is not None`:

Reply:
done